src/wire/ipv6.rs

```diff
@@ -580,6 +580,8 @@ impl fmt::Display for Repr {
 
 use super::pretty_print::{PrettyPrint, PrettyIndent};
 
+// TODO: This is very similar to the implementation for IPv4. Make
+// a way to have less copy and pasted code here.
```

Comment:
In general I'm not too worried about copy/paste in `smoltcp::wire`, various magic to reduce amount of code is an explicit non-goal if you take a look at README.